services/cd-service/pkg/repository/repository.go

```diff
@@ -562,6 +562,11 @@ func (r *repository) ProcessQueueOnce(ctx context.Context, e element, callback P
 	span, ctx := tracer.StartSpanFromContext(e.ctx, "PostPush")
 	defer span.Finish()
 
+	ddError := UpdateDatadogMetrics(r.State(), changes)
```

Comment:
This is the most performance sensitive part of the code that we have. Is that really safe to do? Maybe rather add dedicated Background task for sending and just put this on a bounded queue?

Reply:
as discussed, I added a feature toggle.